fontbe/src/mvar.rs

```diff
@@ -0,0 +1,329 @@
+        assert_eq!(rec.delta_set_outer_index(), 0);
+        assert_eq!(rec.delta_set_inner_index(), 0);
+
+        let Some(Ok(varstore)) = mvar.item_variation_store() else {
+            panic!("MVAR has no ItemVariationStore?!");
+        };
+
+        assert_eq!(varstore.variation_region_list().unwrap().region_count(), 1);
+        assert_eq!(varstore.item_variation_data_count(), 1);
+
+        let vardata = varstore.item_variation_data().get(0).unwrap().unwrap();
+        assert_eq!(vardata.region_indexes(), &[0]);
+        assert_eq!(delta_sets(&vardata), vec![vec![50]]);
+    }
+
+    #[test]
+    fn no_variations_no_mvar() {
+        let regular = NormalizedLocation::for_pos(&[("wght", 0.0)]);
+        let bold = NormalizedLocation::for_pos(&[("wght", 1.0)]);
+        let mut builder = new_mvar_builder(
+            vec![&regular, &bold],
+            vec![axis("wght", 400.0, 400.0, 700.0)],
+        );
+
+        add_sources(&mut builder, "xhgt", &[(&regular, 500.0), (&bold, 500.0)]);
+        add_sources(&mut builder, "cpht", &[(&regular, 800.0), (&bold, 800.0)]);
+
+        // hence no MVAR needed
+        assert!(builder.build().is_none());
+    }
+
+    struct MvarReader<'a> {
+        mvar: read_mvar::Mvar<'a>,
+    }
+
+    impl<'a> MvarReader<'a> {
+        fn new(mvar: read_mvar::Mvar<'a>) -> Self {
+            Self { mvar }
+        }
+
+        fn metric_delta(&self, mvar_tag: &str, coords: &[f32]) -> f64 {
+            let mvar_tag = Tag::from_str(mvar_tag).unwrap();
+            let coords: Vec<F2Dot14> = coords
+                .iter()
+                .map(|coord| F2Dot14::from_f32(*coord))
+                .collect();
+            self.mvar.metric_delta(mvar_tag, &coords).unwrap().to_f64()
+        }
+    }
+
+    #[test]
+    fn sparse_global_metrics() {
+        let regular = NormalizedLocation::for_pos(&[("wght", 0.0)]);
+        let medium = NormalizedLocation::for_pos(&[("wght", 0.5)]);
+        let bold = NormalizedLocation::for_pos(&[("wght", 1.0)]);
+        let mut builder = new_mvar_builder(
+            vec![&regular, &medium, &bold],
+            vec![axis("wght", 400.0, 400.0, 700.0)],
+        );
+        // 'xhgt' defines a value for all three locations
+        add_sources(
+            &mut builder,
+            "xhgt",
+            &[(&regular, 500.0), (&medium, 530.0), (&bold, 550.0)],
+        );
+        // 'strs' is sparse: defines a value for regular and bold, not medium
+        add_sources(&mut builder, "strs", &[(&regular, 50.0), (&bold, 100.0)]);
+
+        let Some(mvar) = builder.build() else {
+            panic!("no MVAR?!");
+        };
+
+        let bytes = dump_table(&mvar).unwrap();
+        let mvar = read_mvar::Mvar::read(FontData::new(&bytes)).unwrap();
+
+        assert_eq!(mvar.value_records().len(), 2);
+        assert_eq!(mvar.value_records()[0].value_tag(), Tag::new(b"strs"));
+        assert_eq!(mvar.value_records()[1].value_tag(), Tag::new(b"xhgt"));
```

Comment:
Optional, why not one assert?

Do you care which index strs and xhgt end up in or just that there is one value record for each of them? If the order doesn't matter perhaps make the test insensitive to it?

Reply:
well, I do care, because they get sorted alphabetically by tag